src/libs/Navigation/linkTo/index.ts

```diff
@@ -249,7 +168,16 @@ export default function linkTo(navigation: NavigationContainerRef<RootStackParam
         }
     }
 
-    if (action && 'payload' in action && action.payload && 'name' in action.payload && isModalNavigator(action.payload.name)) {
+    if (action && 'payload' in action && action.payload && 'name' in action.payload && isSideModalNavigator(action.payload.name)) {
+        // Information about the state may be in the params.
+        const currentFocusedRoute = findFocusedRoute(extrapolateStateFromParams(rootState));
+        const targetFocusedRoute = findFocusedRoute(stateFromPath);
+
+        // If the current focused route is the same as the target focused route, we don't want to navigate.
+        if (currentFocusedRoute?.name === targetFocusedRoute?.name) {
```

Comment:
This line caused a regression - doesn't allow to perform a navigate to the same screen even if the params is different, for example the error param. See https://github.com/Expensify/App/issues/43143